elasticsearch/linux_distributions/opendistro-tar-build.sh

```diff
@@ -13,44 +13,86 @@
-ROOT=$(dirname "$0")
-wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz
-#Untar
-tar -xzf elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz 
-rm -rf elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz
-#Install Plugin
-for plugin_path in opendistro-sql/opendistro_sql-$OD_PLUGINVERSION.zip opendistro-alerting/opendistro_alerting-$OD_PLUGINVERSION.zip opendistro-job-scheduler/opendistro-job-scheduler-$OD_PLUGINVERSION.zip opendistro-security/opendistro_security-$OD_PLUGINVERSION.zip performance-analyzer/opendistro_performance_analyzer-$OD_PLUGINVERSION.zip opendistro-index-management/opendistro_index_management-$OD_PLUGINVERSION.zip opendistro-knn/opendistro-knn-$OD_PLUGINVERSION.zip opendistro-anomaly-detection/opendistro-anomaly-detection-$OD_PLUGINVERSION.zip;
+ARTIFACTS_URL="https://d3g5vo6xdbdb9a.cloudfront.net"
+PACKAGE_NAME="opendistroforelasticsearch"
+TARGET_DIR="$ROOT/target"
+
+# Please DO NOT change the orders, they have dependencies
+PLUGINS="opendistro-sql/opendistro_sql-$OD_VERSION \
+         opendistro-alerting/opendistro_alerting-$OD_VERSION \
+         opendistro-job-scheduler/opendistro-job-scheduler-$OD_VERSION \
+         opendistro-security/opendistro_security-$OD_VERSION \
+         performance-analyzer/opendistro_performance_analyzer-$OD_VERSION \
+         opendistro-index-management/opendistro_index_management-$OD_VERSION \
+         opendistro-knn/opendistro-knn-$OD_VERSION \
+         opendistro-anomaly-detection/opendistro-anomaly-detection-$OD_VERSION"
+
+basedir="${ROOT}/${PACKAGE_NAME}-${OD_VERSION}/plugins"
+PLUGINS_CHECKS="${basedir}/opendistro-job-scheduler \
+                ${basedir}/opendistro_alerting \
+                ${basedir}/opendistro_performance_analyzer \
+                ${basedir}/opendistro_security \
+                ${basedir}/opendistro_sql \
+                ${basedir}/opendistro_index_management \
+                ${basedir}/opendistro-knn \
+                ${basedir}/opendistro-anomaly-detection"
+
+echo $ROOT
+
+if [ -z "$PLUGINS" ]; then
+  echo "Provide plugin list to install (separated by space)"
+  exit 1
+fi
+
+# Prepare target directories
+mkdir ${PACKAGE_NAME}-${OD_VERSION}
+mkdir $TARGET_DIR
+
+# Downloading ES oss
+echo "Downloading ES oss"
+wget -nv https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz ; echo $?
+tar -xzf elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz --strip-components=1 --directory "${PACKAGE_NAME}-${OD_VERSION}" && rm -rf elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz
+# Use tar instead of curl since the image opendistroforelasticsearch/multijava08101112-git:v1 does not have curl
+# And opendistroforelasticsearch/jsenv:v1 does not have java 12
+#curl -Ls https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ES_VERSION-linux-x86_64.tar.gz | tar --strip-components=1 -zxf - --directory "${PACKAGE_NAME}-${OD_VERSION}"
```

Comment:
remove useless comments

Reply:
I keep it here because we possibly move all the codes to scripts.
We will uncomment these sections when that happens.